codegen/lib/graphql_java_gen/templates/APISchema.java.erb

```diff
@@ -266,6 +267,7 @@ public class <%= schema_name %> {
                 <% type.optional_input_fields.each do |field| %>
                   private <%= java_input_type(field.type) %> <%= escape_reserved_word(field.camelize_name) %>;
                 <% end %>
+                  private HashSet<String> fieldsSeen = new HashSet<>();
```

Comment:
This is kind of a weird mix of using generated instance variables and a HashSet.  If we are going to generate instance variables, then we should also generate a boolean to indicate that the field has been set.  If we wanted a sparse object instead, it would make more sense to just use a HashMap, which could use the key's presence to indicate whether or not the field has been set.